Sage/carbon
# refactor: convert Heading to functional component and remove deprecated components

**This PR introduces multiple breaking changes in one to limit number of releases.**

### Proposed behaviour

- Rewrite `Heading` to a functional component.
- Move `RadioButtonMapper` to internal components directory
- Remove following deprecated components:
    - `AppWrapper`
    - `MountInApp`
    - `MultiStepWizard`
    - `ScrollableList`
    - `SubmenuBlock`
- Remove following deprecated props:
    - `as` of `Button`
    - `destructive` of `Confirm`
    - `as` of `Tile`
    - `as` of `Toast`
    - `as` of `SplitButton`
    - `as` of `MultiActionButton`
    - `stickyPosition` and `stickyOffset` of `NavigationBar`

### Current behaviour

- `Heading` is currently class-based.
- `RadioButtonMapper` can be imported by consuming apps - even though it's only intended to be used internally.
- Components and props that have been deprecated for some time are still part of Carbon.

### Checklist




#### QA


### Testing instructions

- Double-check stories for any major regressions.
- Check for any remaining imported/exports that need removing or replacing
